src/main/scala/chisel3/util/Queue.scala

```diff
@@ -185,6 +188,87 @@ object Queue {
     }
   }
 
+  /** A factory for creating shadow queues.  This is created using the
+    * `withShadowQueue` method.
+    */
+  class ShadowFactory[T <: Data] private[Queue] (
```

Comment:
Does this T matter? It seems like it could just be Data...?

Reply:
It looks like it can be dropped. It's perhaps a little more accurate to keep it as the type of `enq` and `deq` really should be the same. However, there's no actually guarantee of this (just because you have one `UInt` doesn't mean that they are the same). I'll drop it.

`ShadowFactory` has a private constructor that is only used by the `withShadow` method. The correct usage is thereby guaranteed.

Nice suggestion, thanks. Changed in a0d582567.